app/controllers/form_controller.rb

```diff
@@ -132,6 +138,10 @@ def successful_redirect_path
         return send("#{@log.model_name.param_key}_#{form.subsection_for_page(@page).id}_check_answers_path", @log)
       end
     end
+    if interruprion_screen_referrer.present?
+      return send("#{@log.class.name.underscore}_#{interruprion_screen_referrer}_path", @log)
```

Comment:
This is potentially dangerous. You're injecting a HTTP header into this method!

Reply:
How about this? [Change how we route back to interruption screen](https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/1584/commits/236a53853f7b2c9920979ff8da43d60f04ce0506)